core/jazz_acl/components/casbin.js

```diff
@@ -17,55 +17,185 @@
 const Enforcer = require('casbin').Enforcer;
 const TypeORMAdapter = require('typeorm-adapter');
 const logger = require('./logger.js');
+const errorHandlerModule = require("./error-handler.js")();
 
 async function dbConnection(config) {
-  const conn = await TypeORMAdapter.default.newAdapter({
-    type: config.CASBIN.TYPE,
-    host: config.CASBIN.HOST,
-    port: config.CASBIN.PORT,
-    username: config.CASBIN.USER,
-    password: config.CASBIN.PASSWORD,
-    database: config.CASBIN.DATABASE,
-    timeout: config.CASBIN.TIMEOUT
-  });
+  let conn;
+  try {
+    conn = await TypeORMAdapter.default.newAdapter({
+      type: config.CASBIN.TYPE,
+      host: config.CASBIN.HOST,
+      port: config.CASBIN.PORT,
+      username: config.CASBIN.USER,
+      password: config.CASBIN.PASSWORD,
+      database: config.CASBIN.DATABASE,
+      timeout: config.CASBIN.TIMEOUT
+    });
+  } catch(err) {
+    if (err.name !== "AlreadyHasActiveConnectionError") {
+      logger.error(err.message);
+      throw (errorHandlerModule.throwInternalServerError(err.message));
+    }
+  }
 
-  const enforcer = await Enforcer.newEnforcer('../config/rbac_model.conf', conn);
-  return enforcer;
+  return conn;
 }
 
-async function addPermissionForUser(serviceId, policies) {
-  //TODO implement addPermissionForUser
-
-  return true;
+/* Add and save the policy in casbin DB*/
+async function addPermissionForUser(serviceId, policies, config) {
```

Comment:
We're making opening/closing DB connections thrice (two times in `removeFilteredPolicy` & one more directly). We should definitely avoid it.